### PR TITLE
completion for properties

### PR DESCRIPTION
**Pull request by Eskibear**
see #843

### CODE REVIEW

File: src/completion/completionProvider.ts [hunk 1]
```diff
@@ -149,6 +150,23 @@ class CompletionProvider implements vscode.CompletionItemProvider {
...
                 return new vscode.CompletionList([snippetItem], false);
             }
+            case XmlTagName.Properties: {
+                const project = mavenExplorerProvider.getMavenProject(document.uri.fsPath);
+                const props = await project?.getProperties();
+
+                if (props) {
+                    const propertyToCompletionItem = (prop: string) => {
+                        const item = new vscode.CompletionItem(prop, vscode.CompletionItemKind.Property);
+                        const insertText = `<${prop}>$1</${prop}>\n$0`;
```
Comment (jdneo) [inline src/completion/completionProvider.ts:160]: Will VS Code automatically correct the EOL according to user's document preference?

Comment (Eskibear) [reply src/completion/completionProvider.ts:160]: Yes, I think so, `\n` is used for all the other snippets and so far so good. See line 28 of this file.

It's a good point and later I'll consider whether to use `os.EOL` instead, for all snippets.

Comment (jdneo) [reply src/completion/completionProvider.ts:160]: You can get it from the input `TextDocument`

Comment (jdneo) [reply src/completion/completionProvider.ts:161]: Change the `\n` here as well?

Comment (Eskibear) [reply src/completion/completionProvider.ts]: updated. also extracted the util function for re-use.

**Review by jdneo — APPROVED** [review]

**Pull request merged**